app/controllers/availabilities_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class AvailabilitiesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @time_ranges = TimeRange.order(:start_time)
+    @availabilities = load_availabilities
+    create_missing_availabilities
+    @start_time = start_time
+    @end_time = end_time
+  end
+
+  def update_availabilities
+    availability_ids = params[:availability_ids] || []
+    Availability.where(user_id: current_user.id).each do |availability|
+      available = availability_ids.include?(availability.id.to_s)
+      availability.update(status: available) if availability.status != available
+    end
+    redirect_to availabilities_path
+  end
+
+  private
+
+  def load_availabilities
+    Hash[Availability.where(user: current_user).joins(:time_range)
+                     .order('day', 'time_ranges.start_time')
+                     .to_a.map do |availability|
```

Comment:
No need the to_a

Reply:
Also, maybe you can abstract this out into a variable, then return Hash[var]? More readable that way.